components/tidb_query/src/codec/data_type/vector.rs

```diff
@@ -218,6 +219,55 @@ impl VectorValue {
         }
     }
 
+    /// Returns maximum encoded size in arrow format.
+    pub fn maximum_encoded_size_arrow(&self, logical_rows: &[usize]) -> Result<usize> {
+        match self {
+            // NullBitMap: (logical_rows.len()/8+1) bytes.  RowCount: 8 bytes.  NullCount: 8 bytes.
+            VectorValue::Int(_) => Ok(logical_rows.len() * 8 + (logical_rows.len() / 8 + 1) + 16),
```

Comment:
It doesn't need to be this accurate, as long as the calculated size > actual size it is ok.

Reply:
Update. @breeswish  PTAL.